doc/SAI-Packet-Action-Group.md

```diff
@@ -0,0 +1,707 @@
+    sai_create_packet_action_group_member_fn        create_packet_action_group_member;
+    sai_remove_packet_action_group_member_fn        remove_packet_action_group_member;
+    sai_set_packet_action_group_member_attribute_fn set_packet_action_group_member_attribute;
+    sai_get_packet_action_group_member_attribute_fn get_packet_action_group_member_attribute;
+
+} sai_packet_action_group_api_t;
+
+```
+
+##### Changes to existing SAI header file
+**In sai.h**
+1. New object for SAI Packet Action Group
+
+```
+    SAI_API_ISOLATION_GROUP     = 40, /**< sai_isolation_group_api_t */
++   SAI_API_PACKET_ACTION_GROUP = 41, /**< sai_packet_action_group_api_t */
+    SAI_API_MAX                 = 42, /**< total number of APIs */
+} sai_api_t;
+
+```
+
+**In saitypes.h**
+1. Add new structure for packet action group actions
+2. Include the new structure in the sai_attribute_value_t structure
+
+```
+/**
+ * @brief Packet Action Group Action Type
+ */
+typedef enum _sai_packet_action_group_action_type_t
+{
+    /** @brief Set Packet Src MAC Address.
+      * @type sai_mac_t
+     */
+    SAI_PACKET_ACTION_GROUP_ACTION_TYPE_SET_SRC_MAC,
+
+    /** @brief Set Packet Dst MAC Address.
+     *  @type sai_mac_t
+     */
+    SAI_PACKET_ACTION_GROUP_ACTION_TYPE_SET_DST_MAC,
+
+    /** @brief Set Packet Outer Vlan Id.
+     *  @type sai_uint16_t
+     */
+    SAI_PACKET_ACTION_GROUP_ACTION_TYPE_SET_OUTER_VLAN_ID,
+
+    /** @brief Set Packet Outer Vlan Priority.
+     *  @type sai_uint8_t
+     */
+    SAI_PACKET_ACTION_GROUP_ACTION_TYPE_SET_OUTER_VLAN_PRI,
+
+    /** @brief Redirect to Port/LAG.
+     *  @type sai_object_id_t
+     */
+    SAI_PACKET_ACTION_GROUP_ACTION_TYPE_REDIRECT,
+
+    /** @brief Decrement TTL.
+     *  @type bool
+     */
+    SAI_PACKET_ACTION_GROUP_ACTION_TYPE_DECREMENT_TTL,
+
+    /** @brief Set Class-of-Service.
+     *  @type sai_uint8_t
+     */
+    SAI_PACKET_ACTION_GROUP_ACTION_TYPE_SET_TC,
+
+    /** @brief Set Packet Color.
+     *  @type sai_uint8_t
+     */
+     SAI_PACKET_ACTION_GROUP_ACTION_TYPE_SET_COLOR,
+
+    /** @brief Set Packet Inner Vlan Id.
+     *  @type sai_uint16_t
+     */
+    SAI_PACKET_ACTION_GROUP_ACTION_TYPE_SET_INNER_VLAN_ID,
+
+    /** @brief Set Packet Inner Vlan Priority.
+     *  @type sai_uint8_t
+     */
+    SAI_PACKET_ACTION_GROUP_ACTION_TYPE_SET_INNER_VLAN_PRI,
+
+    /** @brief Set Packet DSCP.
+     *  @type sai_uint8_t
+     */
+    SAI_PACKET_ACTION_GROUP_ACTION_TYPE_SET_DSCP,
+
+} sai_packet_action_group_action_type_t;
+
+/**
+ * @brief Packet Action Group Action
+ */
+typedef struct _sai_packet_action_group_action_t
+{
+  /**
+   * @brief Action type
+   */
+   sai_packet_action_group_action_type_t action_type;
+
+  /**
+   * @brief Action Value based on the action type
+   */
+   union _action_value {
+     bool booldata;
+     sai_uint8_t u8;
+     sai_uint16_t u16;
+     sai_uint32_t u32;
+     sai_mac_t mac;
+     sai_object_id_t oid;
```

Comment:
having oid here will be hard to implement since all list we assume are no oid members, and we have automated logic to track all oid dependencies but not in nested structs/unions of attr value
i would propose to came out with different approach for oid here

Reply:
@kcudnik Sure once this is discussed in the call, we will think of a different way to solve this.

Reply:
we have a lot of types already in sai_attribute_t and if possible lets keep to them using object attributes, and lets try no to introduce more complex structures

Reply:
We have removed the oid from the structure and made it as an attribute. We want to have a list of packet modification actions in a single attribute and hence we added the structures sai_packet_action_group_action_t and sai_packet_action_group_action_list_t

Reply:
looks good